packages/code-du-travail-api/src/server/routes/items/getRelatedItems.js

```diff
@@ -1,27 +1,96 @@
-const { SOURCES } = require("@socialgouv/cdtn-sources");
+const { SOURCES, getSourceByRoute } = require("@socialgouv/cdtn-sources");
 const { DOCUMENTS } = require("@cdt/data/indexing/esIndexName");
 const { vectorizeQuery } = require("@cdt/data/indexing/vectorizer");
 
 const elasticsearchClient = require("../../conf/elasticsearch.js");
+const getSearchBody = require("./searchBySourceSlug.elastic");
 const getSemBody = require("../search/search.sem");
 const utils = require("../search/utils");
 const getRelatedItemsBody = require("./relatedItems.elastic");
 const { logger } = require("../../utils/logger");
 
-const MAX_RESULTS = 5;
+const MAX_RESULTS = 4;
 
 const ES_INDEX_PREFIX = process.env.ES_INDEX_PREFIX || "cdtn";
 const index = `${ES_INDEX_PREFIX}_${DOCUMENTS}`;
 
-async function getRelatedItems({ title, settings, slug }) {
-  const sources = [
-    SOURCES.TOOLS,
-    SOURCES.SHEET_SP,
-    SOURCES.SHEET_MT,
-    SOURCES.LETTERS,
-    SOURCES.CONTRIBUTIONS,
-    SOURCES.EXTERNALS,
-  ];
+// ratio for A/B testing between covisit and search based related items
+const abRatio = 0.5;
+
+// standard related items :
+const sources = [
+  SOURCES.TOOLS,
+  SOURCES.SHEET_SP,
+  SOURCES.SHEET_MT,
+  SOURCES.LETTERS,
+  SOURCES.CONTRIBUTIONS,
+  SOURCES.EXTERNALS,
+];
+
+// select certain fields and add recommendation source (covisits or search)
+function mapSource(reco, source) {
```

Comment:
I found hard to read / understand the prupose of the function. if you onlu want to add a `reco` prop

why not 
```js
const createAddReco = (reco) => ({
  action,
  description,
  icon,
  reco,
  slug,
  source,
  subtitle,
  title,
  url
}) => ({
  action,
  description,
  icon,
  reco,
  slug,
  source,
  subtitle,
  title,
  url,
  reco,
});
```

Reply:
naïce 👍